Day02/funzioni_day2.py

```diff
@@ -44,17 +44,17 @@ def filter_game_id(partite: dict[int, dict[str, list[int]]]) -> list[int]:
     :return: una lista di game id che soddisfano le condizioni"""
     lista_ids_coretti = []
 
-    lista1 = set(range(12, 21))
-    lista2 = set(range(13, 21))
-    lista3 = set(range(14, 21))
+    lista1 = set(range(13, 21))
+    lista2 = set(range(14, 21))
+    lista3 = set(range(15, 21))
 
     for game_id, colori in partite.items():
         if colori: # Verifica se il dizionario non è vuoto
-            if any(value in lista1 for value in colori.get("red", [])):
+            if any(value > 12 for value in colori.get("red", [])):
                 continue
-            if any(value in lista2 for value in colori.get("green", [])):
+            if any(value > 13 for value in colori.get("green", [])):
                 continue
-            if any(value in lista3 for value in colori.get("blue", [])):
+            if any(value > 14 for value in colori.get("blue", [])):
                 continue
```

Comment:
Questa è un'altra piccola ottimizzazione. Invece di controllare che il valore sia una in una lista di valori validi, possiamo controllare semplicemente che sia più grande della soglia. In questo modo il programma è più veloce e usa meno memoria

Reply:
Con questa piccola modifica le liste con la funzione range non servono più